xarray/core/dataarray.py

```diff
@@ -2233,6 +2233,24 @@ def reset_index(
         ds = self._to_temp_dataset().reset_index(dims_or_levels, drop=drop)
         return self._from_temp_dataset(ds)
 
+    def set_xindex(self, coord_names, index_cls, **kwargs):
```

Comment:
The set_xindex related stuff is not necessary right?

Reply:
that's a artifact from branching off of the `scipy22` branch. I'll rebase at some point, but it *is* useful to create objects that actually contain custom indexes (both for testing and demonstration).

Reply:
Sure is convenient. I just thought it would fit better in a separate PR. 😀

Reply:
`set_xindex` is the temporary API @benbovy wrote for #6783 to enable testing custom indexes, so we probably should never create a PR to `main`. I'll make sure to remove / revert those commits before the PR is merged.

Reply:
xref 6849. I can take care of later opening a separate PR targeting `main` if adding `set_xindex` makes sense.